Sources/CoreCommands/Options.swift

```diff
@@ -217,7 +217,10 @@ public struct SecurityOptions: ParsableArguments {
     #endif
 
     @Option(name: .customLong("resolver-fingerprint-checking"))
-    public var fingerprintCheckingMode: FingerprintCheckingMode = .strict
+    public var fingerprintCheckingMode: WorkspaceConfiguration.CheckingMode = .strict
+
+    @Option(name: .customLong("resolver-signing-entity-checking"))
+    public var signingEntityCheckingMode: WorkspaceConfiguration.CheckingMode = .warn
```

Comment:
These are two different types of checks though--we do fingerprint check for SCM and registry dependency, while signing entity check happens for registry dependency only--and I think we should keep the two options separate, at least initially, because the fingerprint check has been around for some time and is mature enough, while with signing entity check I am more comfortable setting the default level to `warn` because it's new and there might be bugs.

Reply:
Didn't realize there was a fingerprint check for registry dependencies as well, is that a checksum of the archive?

Reply:
> Didn't realize there was a fingerprint check for registry dependencies as well, is that a checksum of the archive?

Yep. We implemented fingerprint for both SCM and registry packages.

Reply:
Does that mean we are grouping two fundamentally different types of fingerprints under one option and is that a potential issue? Registry ones are truly immutable (assuming you can't publish a different archive for an existing version) while SCM ones shouldn't typically change but are technically mutable. 

Let's presume someone is regularly changing their SCM tags after the fact for some reason and decides to change the fingerprint checking policy because of that, as it stands that would affect registry dependencies as well even though it may not be intended?

Reply:
> Does that mean we are grouping two fundamentally different types of fingerprints under one option

Yes, though really, neither of the types should be changing regularly. 

> and is that a potential issue?

I would argue that if a user thinks it's ok to warn instead of error for the uncommon case of regularly changing git hash, then changing checksum is not worse? Checksums are _supposed_ to be immutable, but it depends on registry implementation and thus not guaranteed.

Reply:
I think it's worse in the sense that they accepted an uncommon case and that implicitly opted them into also accepting  a somewhat unrelated case that is likely an actual problem in most scenarios. It's probably fine, though.